### PR TITLE
KAFKA-3453; Transient test failures due to MiniKDC port allocation strategy

### PR DESCRIPTION
**Pull request by ijuma**
Temporarily copy fixed `MiniKdc` class until `MiniKdc` 2.8.0 is released.

### CODE REVIEW

File: core/src/test/java/kafka/security/minikdc/MiniKdc.java [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+            File keytabFile = new File(args[2]).getAbsoluteFile();
+            String[] principals = new String[args.length - 3];
+            System.arraycopy(args, 3, principals, 0, args.length - 3);
+            miniKdc.createPrincipal(keytabFile, principals);
+            System.out.println();
+            System.out.println("Standalone MiniKdc Running");
+            System.out.println("---------------------------------------------------");
+            System.out.println("  Realm           : " + miniKdc.getRealm());
+            System.out.println("  Running at      : " + miniKdc.getHost() + ":" +
+                    miniKdc.getHost());
```
Comment (gwenshap) [inline core/src/test/java/kafka/security/minikdc/MiniKdc.java:171]: While you are there, can we fix this one? it is "  Running at      : " + miniKdc.getHost() + ":" +
-                    miniKdc.getPort()

Comment (ijuma) [inline core/src/test/java/kafka/security/minikdc/MiniKdc.java:171]: Sure, will do.

**Comment by ijuma** [comment]
It has been suggested that it may be better not to depend on MiniKdc at all, so I'll resubmit with that in mind.

**Comment by granthenke** [comment]
@ijuma Tried to find the discussion. Feel free to point me to it if I overlooked it. Just out of curiosity, what is the alternative to MiniKDC?

**Comment by ijuma** [comment]
@granthenke The discussion was offline, I closed the PR to make sure it wasn't merged before I had a chance to resubmit. The reasoning is that the MiniKdc library is a single non-test class (the one I copied in this PR in fact) and it's not worth adding a new dependency for that. I need to take a closer look before I make up my own mind.

**Comment by granthenke** [comment]
Makes sense! Thanks for the info @ijuma

**Comment by granthenke** [comment]
@ijuma Another thing to consider, minikdc is also responsible for [KAFKA-2866](https://issues.apache.org/jira/browse/KAFKA-2866), so I would be happy to get rid of it if we have other options.

**Comment by ijuma** [comment]
Yes, I remember :)

**Pull request closed without merging**